pkg/upgrade/control_plane.go

```diff
@@ -268,7 +289,8 @@ func (u *ControlPlaneUpgrader) Upgrade() error {
 		}
 
 		if err := u.managementClusterClient.Get(context.TODO(), key, &replacement); err != nil {
-			return errors.Wrapf(err, "error getting machine %s", key.String())
+			u.log.Error(err, "error getting machine", "machine", key.String())
```

Comment:
Instead of returning an error here, we just log it and continue. 
If there is a bad replacement machine in the beginning, it is then deleted but the reference of the machine object is still in the slice of `machines`. 
The original behavior would return an error when trying to remove upgrade annotations but we just want it to log it and move on.